plugins/communications-app/ScheduleSection/index.jsx

```diff
@@ -0,0 +1,172 @@
+
+  const checkIsValidSchedule = isScheduled ? scheduleFields : true;
+
+  const handleResetFormValues = () => {
+    dispatch(formActions.resetForm());
+  };
+
+  const handleCloseToast = () => {
+    dispatch(formActions.updateForm({ formStatus: 'default' }));
+  };
+
+  return (
+    <Form.Group>
+      {getConfig().SCHEDULE_EMAIL_SECTION && (
+      <div className="mb-3">
+        <Form.Checkbox
+          name="scheduleEmailBox"
+          checked={isScheduled}
+          onChange={handleChangeScheduled}
+          disabled={formStatus === 'pending'}
+        >
+          {intl.formatMessage(messages.ScheduleSectionSubmitScheduleBox)}
+        </Form.Checkbox>
+      </div>
+      )}
+
+      {isScheduled && (
+      <ScheduleEmailForm
+        isValid={checkIsValidSchedule}
+        onDateTimeChange={handleScheduleDate}
+        dateTime={{ date: scheduleDate, time: scheduleTime }}
+      />
+      )}
+
+      <div
+        className={classNames('d-flex', {
+          'mt-n4.5': !isScheduled && !isMobile,
+          'flex-row-reverse align-items-end': !isMobile,
+          'border-top pt-2': isScheduled,
+        })}
+      >
+
+        {isEditMode && (
+        <Button
+          className="ml-2"
+          variant="outline-brand"
+          onClick={handleResetFormValues}
+        >Cancel
+        </Button>
+        )}
+
+        <StatefulButton
+          className="send-email-btn"
+          data-testid="send-email-btn"
+          variant="primary"
+          onClick={(event) => {
+            event.preventDefault();
+            if (formStatus === 'schedule' && !isScheduledSubmitted) {
+              dispatch(formActions.updateForm({ isScheduleButtonClicked: true }));
+            }
+            openTaskAlert();
+          }}
+          state={formStatus}
+          icons={{
+            default: <Icon src={Send} />,
+            schedule: <Icon src={Event} />,
+            reschedule: <Icon src={Event} />,
+            pending: <Icon src={SpinnerSimple} className="icon-spin" />,
+            complete: <Icon src={Check} />,
+            completeSchedule: <Icon src={Check} />,
+            error: <Icon src={Cancel} />,
+          }}
+          labels={{
+            default: intl.formatMessage(messages.ScheduleSectionSubmitButtonDefault),
+            schedule: intl.formatMessage(messages.ScheduleSectionSubmitButtonSchedule),
+            reschedule: intl.formatMessage(messages.ScheduleSectionSubmitButtonReschedule),
+            pending: intl.formatMessage(messages.ScheduleSectionSubmitButtonPending),
+            complete: intl.formatMessage(messages.ScheduleSectionSubmitButtonComplete),
+            completeSchedule: intl.formatMessage(messages.ScheduleSectionSubmitButtonCompleteSchedule),
+            error: intl.formatMessage(messages.ScheduleSectionSubmitButtonError),
+          }}
+          disabledStates={[
+            'pending',
+            'complete',
+            'completeSchedule',
+          ]}
+        />
```

Comment:
Can we move this into a variables?


```suggestion
          icons={icons}
          labels={labels}
          disabledStates={states}
        />
```